aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1262,6 +1262,50 @@ static inline Tensor & sparse_transpose_(Tensor & self, int64_t dim0, int64_t di
   return self;
 }
 
+Tensor& row_stack_out(Tensor& result, TensorList tensors) {
+  return at::vstack_out(result, tensors);
+}
+
+Tensor row_stack(TensorList tensors) {
+  return at::vstack(tensors);
+}
+
+static std::vector<Tensor> reshape_1d_tensors(TensorList tensors) {
+  std::vector<Tensor> result(tensors.size());
+  auto transform_lambda = [](const Tensor& input) -> Tensor {
+    return input.reshape({input.size(0), 1});
+  };
+  std::transform(tensors.cbegin(),
+                 tensors.cend(),
+                 result.begin(),
+                 transform_lambda);
+  return result;
+}
+
+Tensor& column_stack_out(Tensor& result, TensorList tensors) {
+  TORCH_CHECK(tensors.size() > 0,
+              "column_stack expects a non-empty TensorList");
+
+  auto rep = at::atleast_1d(tensors);
+  if (rep[0].dim() == 1) {
```

Comment:
Is this assuming that every tensor in the list of tensors is 1D, or that every tensor in the list of tensors is 2D? I don't think this has to be the case:

```
a = np.array((1, 2, 3))
b = np.array(((1, 2), (3, 4), (5, 6)))
np.column_stack((a, b))
: array([[1, 1, 2],
       [2, 3, 4],
       [3, 5, 6]])
```

Reply:
```
one = np.array((-1, -2, -3))
two = np.array(((1, 2), (3, 4), (5, 6)))
result_0 = np.column_stack((one, two, two))
result_1 = np.hstack((one.reshape(3, 1), two, two))
result_0 == result_1
: array([[ True,  True,  True,  True,  True],
       [ True,  True,  True,  True,  True],
       [ True,  True,  True,  True,  True]])
```

I think the transform just needs to be made conditional on the tensor being 1D?

Reply:
Thanks so much for the detailed explanations, the transform here only needs to be applied to 1D tensors.